src/zui/ZUIDuration/index.tsx

```diff
@@ -22,26 +27,100 @@ type DurField = {
   visible: boolean;
 };
 
+const timeUnitMap = new Map([
+  ['milliseconds', 0],
+  ['seconds', 1],
+  ['minutes', 2],
+  ['hours', 3],
+  ['days', 4],
+] as const);
+
 const ZUIDuration: FC<Props> = ({
-  withDays = true,
-  withHours = true,
-  withThousands = false,
-  withMinutes = true,
-  withSeconds = false,
+  upperTimeUnit = 'days',
+  lowerTimeUnit = 'minutes',
   seconds,
 }) => {
-  const ms = (seconds * 1000) % 1000;
-  const s = Math.floor(seconds) % 60;
-  const m = Math.floor(seconds / 60) % 60;
-  const h = Math.floor(seconds / 60 / 60) % 24;
-  const days = Math.floor(seconds / 60 / 60 / 24);
+  let upper = timeUnitMap.get(upperTimeUnit) ?? 4;
+  const lower = timeUnitMap.get(lowerTimeUnit) ?? 2;
+
+  if (upper < lower) {
+    upper = lower;
+  }
```

Comment:
I think this is a fine solution! The only other options I can think of are:

1. Throw errors when the props are invalid (bad because could happen at runtime)
2. Utilize the type-system somehow, e.g. by having a single property with an enum (or string union type) with values for every possible combination (and no values for impossible combinations)

I think option 2 might be nice, but I haven't thought about how many possible combinations there are (10?).